internal/signer/darwin/keychain/keychain.go

```
@@ -405,3 +443,123 @@ func certIn(xc *x509.Certificate, xcs []*x509.Certificate) bool {
+	algorithms, algoErr := k.getEncryptAlgorithm()
+	if algoErr != nil {
+		fmt.Printf("algorithm is unsupported. only RSA algorithms are supported. %v", algoErr)
+	}
+	// Each padding scheme has varying number of bytes.
+	pssPaddingBytes := 20
+	oaepPaddingBytes := 130
+	pkcsPaddingBytes := 11
+	switch algorithms {
+	case C.kSecKeyAlgorithmRSASignatureDigestPSSSHA256,
+		C.kSecKeyAlgorithmRSASignatureDigestPSSSHA384,
+		C.kSecKeyAlgorithmRSASignatureDigestPSSSHA512:
+		return pssPaddingBytes
+	case C.kSecKeyAlgorithmRSAEncryptionOAEPSHA256,
+		C.kSecKeyAlgorithmRSAEncryptionOAEPSHA384,
+		C.kSecKeyAlgorithmRSAEncryptionOAEPSHA512:
+		return oaepPaddingBytes
+	case C.kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA256,
+		C.kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA384,
+		C.kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA512:
+		return pkcsPaddingBytes
+	default:
+		return int(UNKNOWN_SECKEY_ALGORITHM)
+	}
+}
+
+func (k *Key) checkDataSize(plaintext []byte) error {
+	// Plaintext data must be smaller than the key's block size minus padding space.
+	sizeLim := uint64(C.SecKeyGetBlockSize(k.publicKeyRef)) - uint64(k.getPaddingSize())
+	if uint64(len(plaintext)) >= sizeLim {
+		return fmt.Errorf("plaintext is too long")
+	}
+	return nil
+}
+
+func (k *Key) getRSAEncryptAlgorithm() (C.SecKeyAlgorithm, error) {
+	var algorithms map[crypto.Hash]C.CFStringRef
+	switch pub := k.Public().(type) {
+	case *rsa.PublicKey:
+		if C.SecKeyIsAlgorithmSupported(k.publicKeyRef, C.kSecKeyOperationTypeEncrypt, C.kSecKeyAlgorithmRSASignatureDigestPSSSHA256) == 1 {
+			algorithms = rsaPSSAlgorithms
+		} else if C.SecKeyIsAlgorithmSupported(k.publicKeyRef, C.kSecKeyOperationTypeEncrypt, C.kSecKeyAlgorithmRSAEncryptionOAEPSHA256) == 1 {
+			algorithms = rsaOAEPAlgorithms
+		} else if C.SecKeyIsAlgorithmSupported(k.publicKeyRef, C.kSecKeyOperationTypeEncrypt, C.kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA256) == 1 {
+			algorithms = rsaPKCS1v15Algorithms
+		}
```

the else statement should return an error, saying unknown RSA argument.